spring-cloud-kubernetes-commons/src/main/java/org/springframework/cloud/kubernetes/commons/config/KubernetesConfigDataLoaderLoggerReconfigurer.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright 2013-2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.kubernetes.commons.config;
+
+import java.lang.invoke.MethodHandles;
+import java.lang.invoke.MethodType;
+import java.lang.reflect.Modifier;
+import java.util.Arrays;
+
+import org.apache.commons.logging.Log;
+
+import org.springframework.boot.logging.DeferredLogFactory;
+import org.springframework.cloud.kubernetes.client.config.K8sNativeKubernetesConfigDataLoggerHidden;
+import org.springframework.cloud.kubernetes.fabric8.config.Fabric8KubernetesConfigDataLoggerHidden;
+import org.springframework.core.io.Resource;
+import org.springframework.core.io.support.PathMatchingResourcePatternResolver;
+import org.springframework.util.ClassUtils;
+
+/**
+ *
+ * See :
+ * <a href="https://github.com/spring-cloud/spring-cloud-kubernetes/issues/1173">this
+ * issue</a>.
+ *
+ * @author wind57
+ */
+final class KubernetesConfigDataLoaderLoggerReconfigurer {
+
+	private static final String COMMONS_PACKAGE = "org/springframework/cloud/kubernetes/commons/config";
+
+	private static final String FABRIC8_PACKAGE = "org/springframework/cloud/kubernetes/fabric8/config";
+
+	private static final String K8S_NATIVE_PACKAGE = "org/springframework/cloud/kubernetes/client/config";
+
+	private static final PathMatchingResourcePatternResolver RESOLVER = new PathMatchingResourcePatternResolver();
+
+	private static Log log;
+
+	private KubernetesConfigDataLoaderLoggerReconfigurer() {
+	}
+
+	static void reconfigureLoggers(DeferredLogFactory logFactory) {
+
+		log = logFactory.getLog(KubernetesConfigDataLoaderLoggerReconfigurer.class);
+
+		try {
+			Resource[] commonResources = RESOLVER.getResources("classpath*:" + COMMONS_PACKAGE + "/*.class");
```

Comment:
the idea here is this:

- let's scan for all classes in the packages we care about, where potentially there are logger instances related to our config-data logic. 
- Once I have one class, create a `MethodHandles.Lookup privateLookup = MethodHandles.privateLookupIn(cls, MethodHandles.lookup());`. Basically in simple words give me a Lookup that can access private members.
- Once I have the above lookup, create a "hidden class" that has `NESTMATE` access to the class where the logger is. Think about it like this: If I found `ConfigUtils` for example, create another class (hidden), that has access to private members of `ConfigUtils`. Another way to think about it is an inner class inside `ConfigUtils`. 
- Once I create that hidden class, get its Nest Host and find a method that is static, and takes a Log.class type as a parameter. Once that method is found, get its name and invoke it with a logger instance that I already created.

---

If even more simplified : If I want to override the logger in `ConfigUtils`, then I need to create a method inside it : 

```
	// called from config-data loader
	private static void logger(Log logger) {
		ConfigUtils.logger = logger;
	}
```

Then I create an inner class inside `ConfigUtils` (not really, it's called a NESTMATE, but to build the mental model, I guess it works). With this inner class, call `getNestHost` (`ConfigUtils` that is) and invoke that `logger` method with the argument that I will provide.